### PR TITLE
Fixed issue #14290: Some form can not be submitted if csrfTokenName is updated

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: adding csrfTokenName for input name in LS.data
Dev: adding csrfTokenData as object to allow usage of $.extend in LS.data
Dev: replace 'YII_CSRF_TOKEN' by LS.data.csrfTokenName when found
Dev: usage of $.extend for uploader (public)
Dev: but **have a syntax error in gulp …**

### CODE REVIEW

**Comment by Shnoulle** [comment]
@lacrioque , just updated to Fedora29, can youn help me here please ?

```
[shnoulle@poledra adminbasics]$ gulp default
[11:15:06] Using gulpfile /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/gulpfile.js
[11:15:06] Starting 'default'...
[11:15:06] Starting 'compile:production'...
[11:15:06] Starting 'sass:production'...
[11:15:06] Starting 'webpack:production'...
[11:15:06] Finished 'sass:production' after 158 ms
[11:15:07] Version: webpack 3.12.0
             Asset     Size  Chunks                    Chunk Names
    adminbasics.js   870 kB       0  [emitted]  [big]  main
adminbasics.js.map  1.12 MB       0  [emitted]         main
[11:15:07] Finished 'webpack:production' after 869 ms
[11:15:07] Starting 'compress'...
[BABEL] Note: The code generator has deoptimised the styling of "/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js" as it exceeds the max of "500KB".
[11:15:13] Finished 'compress' after 6.69 s
[11:15:13] Starting 'compresslibs'...
[11:15:13] Finished 'compresslibs' after 4.48 ms
[11:15:13] Finished 'compile:production' after 7.57 s
[11:15:13] Starting 'sass'...
[11:15:13] Starting 'webpack'...
[11:15:13] Starting 'babelify'...
[11:15:13] Finished 'sass' after 57 ms
[11:15:14] Version: webpack 3.12.0
             Asset     Size  Chunks                    Chunk Names
    adminbasics.js   870 kB       0  [emitted]  [big]  main
adminbasics.js.map  1.12 MB       0  [emitted]         main
[11:15:15] 'babelify' errored after 1.11 s
[11:15:15] SyntaxError in plugin "gulp-babel"
Message:
    /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js: Unexpected token, expected ( (26552:9)
[11:15:15] 'default' errored after 8.69 s
[11:15:15] SyntaxError in plugin "run-sequence(babelify)"
Message:
    /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js: Unexpected token, expected ( (26552:9)
Details:
    pos: 819195
    loc: [object Object]
    _babel: true
    codeFrame:   26550 |       }
  26551 | 
> 26552 |       if 
        |          ^
    fileName: /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js

Stack:
SyntaxError: /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js: Unexpected token, expected ( (26552:9)
  26550 |       }
  26551 | 
> 26552 |       if 
        |          ^
    at Parser.pp$5.raise (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:4454:13)
    at Parser.pp.unexpected (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:1761:8)
    at Parser.pp.expect (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:1749:33)
    at Parser.pp$3.parseParenExpression (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:3788:8)
    at Parser.pp$1.parseIfStatement (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:2058:20)
    at Parser.pp$1.parseStatement (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:1846:19)
    at Parser.pp$1.parseBlockBody (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:2268:21)
    at Parser.pp$1.parseBlock (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:2247:8)
    at Parser.pp$3.parseFunctionBody (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:4235:22)
    at Parser.pp$1.parseFunction (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:2386:8)
[11:15:15] Finished 'webpack' after 1.12 s
```

File: assets/scripts/admin/assessments.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ var bindAction = function(){
         $('input[name=action]').val('assessmentupdate');
         $.ajax({
             url: loadEditUrl,
-            data: {id: $(this).closest('tr').data('assessment-id'), YII_CSRF_TOKEN : LS.data.csrfToken},
+            data: {id: $(this).closest('tr').data('assessment-id')},// crsf is already in ajaxsetup
```
Comment (Shnoulle) [inline assets/scripts/admin/assessments.js]: But more : it's a GET request : no need csrfToken

**Comment by Shnoulle** [comment]
@lacrioque 

Currently i have an error on master too : 

```
[shnoulle@poledra adminbasics]$ gulp default
[14:44:51] Using gulpfile /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/gulpfile.js
[14:44:51] Starting 'default'...
[14:44:51] Starting 'compile:production'...
[14:44:51] Starting 'sass:production'...
[14:44:51] Starting 'webpack:production'...
[14:44:52] Finished 'sass:production' after 187 ms
[14:44:53] Version: webpack 3.12.0
             Asset     Size  Chunks                    Chunk Names
    adminbasics.js   870 kB       0  [emitted]  [big]  main
adminbasics.js.map  1.12 MB       0  [emitted]         main
[14:44:53] Finished 'webpack:production' after 1.11 s
[14:44:53] Starting 'compress'...
[BABEL] Note: The code generator has deoptimised the styling of "/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js" as it exceeds the max of "500KB".
[14:44:59] Finished 'compress' after 6.83 s
[14:44:59] Starting 'compresslibs'...
[14:44:59] Finished 'compresslibs' after 4.56 ms
[14:44:59] Finished 'compile:production' after 7.95 s
[14:44:59] Starting 'sass'...
[14:44:59] Starting 'webpack'...
[14:44:59] Starting 'babelify'...
[14:44:59] Finished 'sass' after 50 ms
[14:45:00] Version: webpack 3.12.0
             Asset     Size  Chunks                    Chunk Names
    adminbasics.js   870 kB       0  [emitted]  [big]  main
adminbasics.js.map  1.12 MB       0  [emitted]         main
[14:45:00] 'babelify' errored after 1.01 s
[14:45:00] SyntaxError in plugin "gulp-babel"
Message:
    /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js: Unexpected token (26260:14)
[14:45:00] 'default' errored after 8.98 s
[14:45:00] SyntaxError in plugin "run-sequence(babelify)"
Message:
    /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js: Unexpected token (26260:14)
Details:
    pos: 811003
    loc: [object Object]
    _babel: true
    codeFrame:   26258 |           maybeInvokeDelegate(delegate, context);
  26259 | 
> 26260 |           if (
        |               ^
    fileName: /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js

Stack:
SyntaxError: /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js: Unexpected token (26260:14)
  26258 |           maybeInvokeDelegate(delegate, context);
  26259 | 
> 26260 |           if (
        |               ^
    at Parser.pp$5.raise (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:4454:13)
    at Parser.pp.unexpected (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:1761:8)
    at Parser.pp$3.parseExprAtom (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:3750:12)
    at Parser.pp$3.parseExprSubscripts (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:3494:19)
    at Parser.pp$3.parseMaybeUnary (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:3474:19)
    at Parser.pp$3.parseExprOps (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:3404:19)
    at Parser.pp$3.parseMaybeConditional (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:3381:19)
    at Parser.pp$3.parseMaybeAssign (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:3344:19)
    at Parser.pp$3.parseExpression (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:3306:19)
    at Parser.pp$3.parseParenExpression (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:3789:18)
[14:45:00] Finished 'webpack' after 1.02 s
```

```
[shnoulle@poledra adminbasics]$ gulp default --no-color
[14:45:40] Using gulpfile /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/gulpfile.js
[14:45:40] Starting 'default'...
[14:45:40] Starting 'compile:production'...
[14:45:40] Starting 'sass:production'...
[14:45:40] Starting 'webpack:production'...
[14:45:40] Finished 'sass:production' after 163 ms
[14:45:41] Version: webpack 3.12.0
             Asset     Size  Chunks                    Chunk Names
    adminbasics.js   870 kB       0  [emitted]  [big]  main
adminbasics.js.map  1.12 MB       0  [emitted]         main
[14:45:41] Finished 'webpack:production' after 943 ms
[14:45:41] Starting 'compress'...
[BABEL] Note: The code generator has deoptimised the styling of "/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js" as it exceeds the max of "500KB".
[14:45:48] Finished 'compress' after 6.8 s
[14:45:48] Starting 'compresslibs'...
[14:45:48] Finished 'compresslibs' after 3.99 ms
[14:45:48] Finished 'compile:production' after 7.75 s
[14:45:48] Starting 'sass'...
[14:45:48] Starting 'webpack'...
[14:45:48] Starting 'babelify'...
[14:45:48] Finished 'sass' after 54 ms
[14:45:48] Version: webpack 3.12.0
             Asset     Size  Chunks                    Chunk Names
    adminbasics.js   870 kB       0  [emitted]  [big]  main
adminbasics.js.map  1.12 MB       0  [emitted]         main
[14:45:49] 'babelify' errored after 817 ms
[14:45:49] SyntaxError in plugin "gulp-babel"
Message:
    /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js: Unexpected token, expected ) (26552:14)
[14:45:49] 'default' errored after 8.58 s
[14:45:49] SyntaxError in plugin "run-sequence(babelify)"
Message:
    /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js: Unexpected token, expected ) (26552:14)
Details:
    pos: 819195
    loc: [object Object]
    _babel: true
    codeFrame:   26550 |       }
  26551 | 
> 26552 |       if (fina
        |               ^
    fileName: /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js

Stack:
SyntaxError: /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js: Unexpected token, expected ) (26552:14)
  26550 |       }
  26551 | 
> 26552 |       if (fina
        |               ^
    at Parser.pp$5.raise (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:4454:13)
    at Parser.pp.unexpected (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:1761:8)
    at Parser.pp.expect (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:1749:33)
    at Parser.pp$3.parseParenExpression (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:3790:8)
    at Parser.pp$1.parseIfStatement (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:2058:20)
    at Parser.pp$1.parseStatement (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:1846:19)
    at Parser.pp$1.parseBlockBody (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:2268:21)
    at Parser.pp$1.parseBlock (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:2247:8)
    at Parser.pp$3.parseFunctionBody (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:4235:22)
    at Parser.pp$1.parseFunction (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:2386:8)
[14:45:49] Finished 'webpack' after 835 ms
```
Did you think it's an issue in gulp version on Fedora only or an issue with gulp version related to an issue with our config ?

**Comment by Shnoulle** [comment]
```
$ gulp -v
[14:51:43] CLI version 2.0.1
[14:51:43] Local version 3.9.1
```

**Comment by Shnoulle** [comment]
```
[shnoulle@poledra adminbasics]$ npm -v
6.4.1
[shnoulle@poledra adminbasics]$ gulp -v
[15:35:48] CLI version 2.0.1
[15:35:48] Local version 3.9.1
```

**Comment by Shnoulle** [comment]
Not fixed …

```
[shnoulle@poledra adminbasics]$ npm -v
6.4.1
[shnoulle@poledra adminbasics]$ gulp -v
[15:35:48] CLI version 2.0.1
[15:35:48] Local version 3.9.1
[shnoulle@poledra adminbasics]$ node -v
v10.13.0
[shnoulle@poledra adminbasics]$ npm i eslint-plugin-import eslint-plugin-node --save
npm WARN adminbasics No description
npm WARN adminbasics No repository field.
npm WARN adminbasics No license field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

+ eslint-plugin-import@2.14.0
+ eslint-plugin-node@8.0.0
added 17 packages from 7 contributors and audited 11349 packages in 6.186s
found 5 vulnerabilities (1 low, 4 high)
  run `npm audit fix` to fix them, or `npm audit` for details
[shnoulle@poledra adminbasics]$ gulp default
[15:37:52] Using gulpfile /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/gulpfile.js
[15:37:52] Starting 'default'...
[15:37:52] Starting 'compile:production'...
[15:37:52] Starting 'sass:production'...
[15:37:52] Starting 'webpack:production'...
[15:37:53] Finished 'sass:production' after 329 ms
[15:37:53] Version: webpack 3.12.0
             Asset     Size  Chunks                    Chunk Names
    adminbasics.js   870 kB       0  [emitted]  [big]  main
adminbasics.js.map  1.12 MB       0  [emitted]         main
[15:37:53] Finished 'webpack:production' after 1.07 s
[15:37:53] Starting 'compress'...
[BABEL] Note: The code generator has deoptimised the styling of "/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js" as it exceeds the max of "500KB".
[15:38:00] Finished 'compress' after 6.82 s
[15:38:00] Starting 'compresslibs'...
[15:38:00] Finished 'compresslibs' after 3.39 ms
[15:38:00] Finished 'compile:production' after 7.9 s
[15:38:00] Starting 'sass'...
[15:38:00] Starting 'webpack'...
[15:38:00] Starting 'babelify'...
[15:38:00] Finished 'sass' after 56 ms
[15:38:01] Version: webpack 3.12.0
             Asset     Size  Chunks                    Chunk Names
    adminbasics.js   870 kB       0  [emitted]  [big]  main
adminbasics.js.map  1.12 MB       0  [emitted]         main
[15:38:01] 'babelify' errored after 1.03 s
[15:38:01] SyntaxError in plugin "gulp-babel"
Message:
    /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js: Unexpected token (25810:31)
[15:38:01] 'default' errored after 8.94 s
[15:38:01] SyntaxError in plugin "run-sequence(babelify)"
Message:
    /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js: Unexpected token (25810:31)
Details:
    pos: 794619
    loc: [object Object]
    _babel: true
    codeFrame:   25808 | 
  25809 | // ie9- setTimeout & setInterval additional parameters fix
> 25810 | var global = __webpack_require_
        |                                ^
    fileName: /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js

Stack:
SyntaxError: /mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/build/adminbasics.js: Unexpected token (25810:31)
  25808 | 
  25809 | // ie9- setTimeout & setInterval additional parameters fix
> 25810 | var global = __webpack_require_
        |                                ^
    at Parser.pp$5.raise (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:4454:13)
    at Parser.pp.unexpected (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:1761:8)
    at Parser.pp$3.parseExprAtom (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:3750:12)
    at Parser.pp$3.parseExprSubscripts (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:3494:19)
    at Parser.pp$3.parseMaybeUnary (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:3474:19)
    at Parser.pp$3.parseExprOps (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:3404:19)
    at Parser.pp$3.parseMaybeConditional (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:3381:19)
    at Parser.pp$3.parseMaybeAssign (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:3344:19)
    at Parser.pp$3.parseExpression (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:3306:19)
    at Parser.pp$1.parseStatement (/mnt/data/shnoulle/nginx/www/master/assets/packages/adminbasics/node_modules/babylon/lib/index.js:1906:19)
[15:38:01] Finished 'webpack' after 1.05 s
```

**Comment by lacrioque** [comment]
Merged recompiling it without issues

**Comment by Shnoulle** [comment]
Thanks :)

**Pull request merged**